### ISSUE TITLE
Update AMP compatibility logic

### ISSUE THREAD

**Issue by swissspidy** [swissspidy/gutenberg-bento#33]
The AMP plugin continues to improve its support for Bento, see e.g. https://github.com/ampproject/amp-wp/pull/6722

I want to make sure this example repo works fine with that.

Not sure what's needed there yet.

**Comment by swissspidy** [comment]
cc @westonruter

**Comment by westonruter** [comment]
The intention is that if you're using `bento`-prefixed components that it should automatically get converted. Your `AMP_Carousel_Sanitizer` is searching for `bento-base-carousel` components and that is fine because the conversion to `amp-base-carousel` would happen after that 3rd-party sanitizer.

Otherwise, you're currently omitting the custom Bento scripts when `is_amp` so the AMP plugin won't trip over them. But in the case where you're not self-hosting, the aforementioned PR would automatically identify the scripts and stylesheets for Bento components and automatically remove them since the plugin's auto-extension importing would add the AMP versions.

As I refine my PR I'll check to make sure it's working as expected with your plugin.

**Comment by swissspidy** [comment]
I think the script registration part would also need updating:

https://github.com/swissspidy/gutenberg-bento/blob/9b56b8237b4736f4998ad40a9e4d98c70223bc97/inc/namespace.php#L65-L69

https://github.com/swissspidy/gutenberg-bento/blob/9b56b8237b4736f4998ad40a9e4d98c70223bc97/inc/namespace.php#L74-L79

I wouldn't expect the AMP plugin to register any Bento assets.

**Comment by westonruter** [comment]
Right, but what the new Bento sanitizer does in my PR is it _removes_ those Bento-specific scripts and stylesheets.

Nevertheless, the scripts at least won't be enqueued anyway on AMP because of the `is_amp()` check here:

https://github.com/swissspidy/gutenberg-bento/blob/9b56b8237b4736f4998ad40a9e4d98c70223bc97/inc/namespace.php#L205-L207

**Comment by westonruter** [comment]
I just tested with https://github.com/ampproject/amp-wp/pull/6722 and it seems to work just fine. On AMP pages, the `bento-base-carousel` is converted to `amp-base-carousel` and there are no validation errors raised.

**Comment by westonruter** [comment]
With the latest changes in that PR, if you add `style="aspect-ratio: 4/1;"` to the `bento-base-accordion` you can eliminate these lines:

https://github.com/swissspidy/gutenberg-bento/blob/42e34904a3b9d639770e9ceb2193ec0fa2153fb6/inc/class-amp-carousel-sanitizer.php#L50-L54

The `AMP_Bento_Sanitizer` will look at inline styles on Bento elements and convert them into the required AMP layout attributes, where possible.

**Comment by swissspidy** [comment]
Are you suggesting adding that inline style in this sanitizer, or just generally? Right now the aspect ratio is set in the CSS file already

**Comment by westonruter** [comment]
Actually rather that the inline `style` attribute could be output in the `save` function.

**Comment by swissspidy** [comment]
Makes sense. Just feels a tiny bit redundant as there's already an aspect ratio declaration here:

https://github.com/swissspidy/gutenberg-bento/blob/42e34904a3b9d639770e9ceb2193ec0fa2153fb6/src/carousel/view.css#L1-L6

**Comment by westonruter** [comment]
Right. It doesn't have to be inlined. But it's an alternative that would reduce sanitizer logic. Definitely not required. But then again, somewhat better to not have the ratio defined in two places. What if you wanted to add an aspect ratio setting to the block too?

**Comment by swissspidy** [comment]
Thet is a good point 👍
